build/check-version.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+import argparse
+import functools
+
+
+@functools.total_ordering
+class SemVer:
+    def __init__(self, version_str: str):
+        try:
+            major, minor, patch = version_str.split(".", 3)
+
+            self.major = int(major)
+            self.minor = int(minor)
+            self.patch = int(patch)
+        except ValueError as e:
+            raise ValueError(f"Integers in form MAJOR.MINOR.PATCH format expected, received {version_str}") from e
+
+    def __eq__(self, other: "SemVer"):
+        return self.major == other.major and \
+               self.minor == other.minor and \
+               self.patch == other.patch
+
+    def __lt__(self, other: "SemVer"):
+        return self != other and \
+               self.major <= other.major and \
+               self.minor <= other.minor and \
+               self.patch <= other.patch
+
+    def __repr__(self):
+        return f"SemVer('{self.major}.{self.minor}.{self.patch}"
```

Comment:
```suggestion
        return f"SemVer('{self.major}.{self.minor}.{self.patch})"
```

Reply:
```suggestion
        return f"SemVer('{self.major}.{self.minor}.{self.patch}')"
```
Good spot, but I think this is what we both mean